### PR TITLE
Move service_util endpoints related functions to framework/endpoints/ports.go

### PR DESCRIPTION
**Pull request by jiatongw**
**What type of PR is this?**

/kind cleanup
/priority backlog

**What this PR does / why we need it**:

- Move service_util.go endpoints related functions to framework/endpoints/ports.go.
- Fix references of some methods to use the right package

Reference issue #77197, #76206 (parent issue #75601 )

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/kubectl/kubectl.go
Comment (jiatongw) [inline test/e2e/kubectl/kubectl.go]: Rename variable `endpoints` to `ep` because we need to import `endpoints` package in this file

**Comment by neolit123** [comment]
thanks for the PR @jiatongw 
/approve
/ok-to-test

File: test/e2e/network/service.go
Comment (timothysc) [inline test/e2e/network/service.go]: given the other naming patterns we should prefix with e2e to avoid collisions with existing libraries.

**Review by timothysc — DISMISSED** [review]
minor nits then lgtm

File: test/e2e/network/service.go
Comment (jiatongw) [reply test/e2e/network/service.go]: `e2eendpoints` is a bit weird.. do you have some suggestions?

Comment (neolit123) [reply test/e2e/network/service.go]: i think it's fine. the double `ee` is not a problem.

**Comment by timothysc** [comment]
/approve 

to unblock post renaming in case others want to lgtm earlier.

File: test/e2e/network/service.go
Comment (jiatongw) [reply test/e2e/network/service.go]: Thanks Lubomir

**Comment by jiatongw** [comment]
There are conflicts because we both modified `service_util.go`. I integrated both yours and mine comments on `PortsByPodName`, `PortsByPodUID`, `GetContainerPortsByPodUID` and `ValidateEndpointsOrFail`, which in `ports.go`.  Please kindly review @atoato88

**Comment by jiatongw** [comment]
/assign @oomichi 

Assign you as well since you are the approver of PR #77643. Please kindly review :)

File: test/e2e/framework/endpoints/ports.go
Comment (oomichi) [inline test/e2e/framework/endpoints/ports.go]: Should we remove this commented-out line?

File: test/e2e/framework/endpoints/ports.go
Comment (oomichi) [inline test/e2e/framework/endpoints/ports.go]: ditto

File: test/e2e/framework/endpoints/ports.go
Comment (oomichi) [inline test/e2e/framework/endpoints/ports.go]: ditto

**Review by oomichi — COMMENTED** [review]
/test pull-kubernetes-e2e-gce-device-plugin-gpu

File: test/e2e/framework/endpoints/ports.go
Comment (andrewsykim) [inline test/e2e/framework/endpoints/ports.go]: Maybe for a future PR, we should try to avoid importing `framework` here. Maybe a better pattern is to return an err here and have the calling test run `framework.Failf`.

File: test/e2e/framework/endpoints/ports.go
Comment (jiatongw) [reply test/e2e/framework/endpoints/ports.go]: Those code are already commented out before I opened this PR so I don't touch them.
I think they are good to remove since we are more interested in error logs. I'll delete those lines. Thanks

Comment (andrewsykim) [reply test/e2e/framework/endpoints/ports.go]: Looks like these comments were here before, agreed that they're probably safe to remove

File: test/e2e/framework/endpoints/ports.go
Comment (andrewsykim) [inline test/e2e/framework/endpoints/ports.go]: Can we rename this to `ValidateEndpointsPortsOrFail`?

File: test/e2e/framework/endpoints/ports.go
Comment (andrewsykim) [reply test/e2e/framework/endpoints/ports.go]: Actually, if you don't mind I think it would be ideal to do this refactor now. It'll keep the dependency graph cleaner for later

File: test/e2e/framework/endpoints/BUILD
Comment (andrewsykim) [inline test/e2e/framework/endpoints/BUILD]: I thought we got rid of this file?

**Comment by jiatongw** [comment]
Keep 2 commits for now, if good to merge, I'll squash them to 1 commit

File: test/e2e/framework/endpoints/ports.go
Comment (jiatongw) [reply test/e2e/framework/endpoints/ports.go]: updated, ready to review

File: test/e2e/framework/endpoints/ports.go
Comment (figo) [reply test/e2e/framework/endpoints/ports.go]: > Keep 2 commits for now, if good to merge, I'll squash them to 1 commit

it is better to squash if just two commits and very small.

**Comment by figo** [comment]
> Keep 2 commits for now, if good to merge, I'll squash them to 1 commit

it is better to squash if just two commits and very small.

File: test/e2e/framework/endpoints/ports.go
Comment (andrewsykim) [inline test/e2e/framework/endpoints/ports.go]: If `framework.ServiceStartTimeout` is the only reason we're importing `framework` here I think it would be okay to duplicate this constant into this package or import it somewhere else. What do you think?

**Comment by atoato88** [comment]
@jiatongw Thank you for mention. I've confirmed that your comments on next items are OK for me. :smiley: 
- PortsByPodName
- PortsByPodUID
- GetContainerPortsByPodUID
- ValidateEndpointsPortsOrFail (formerly ValidateEndpointsOrFail)

File: test/e2e/framework/endpoints/ports.go
Comment (jiatongw) [reply test/e2e/framework/endpoints/ports.go]: Yes, I think it's ok to do that. Thanks! Updated.

**Pull request merged**
